frontend/test/resolution/testManage.cpp

```diff
@@ -0,0 +1,265 @@
+      record R : contextManager {
+        proc type contextReturnType type do return int;
+
+        proc enterContext() {
+          return global;
+        }
+        proc enterContext() ref {
+          return global;
+        }
+        proc enterContext() const ref {
+          return global;
+        }
+        proc exitContext(in error: owned Error?) {}
+      }
+      manage new R() as byVal, new R() as ref byRef, new R() as const ref byConstRef {
+
+      }
+    }
+    )""";
+
+  setFileText(ctx, "test.chpl", program);
+  auto& modules = parsing::parse(ctx, UniqueString::get(ctx, "test.chpl"), UniqueString());
+  assert(modules.size() == 1);
+  auto& rr = resolveModule(ctx, modules[0]->id());
+
+  bool found;
+
+  auto byVal = findVariable(modules[0], "byVal");
+  auto byValParent = parsing::parentAst(ctx, byVal)->toAs()->symbol();
+  assert(byValParent);
+  assert(rr.hasAst(byValParent));
+  found = false;
+  for (auto aa : rr.byAst(byValParent).associatedActions()) {
+    if (aa.action() == AssociatedAction::ENTER_CONTEXT) {
+      found = true;
+      assert(aa.fn()->id().symbolPath() == "M.R.enterContext");
+    }
+  }
+  assert(found);
+
+  auto byRef = findVariable(modules[0], "byRef");
+  auto byRefParent = parsing::parentAst(ctx, byRef)->toAs()->symbol();
+  assert(byRefParent);
+  assert(rr.hasAst(byRefParent));
+  found = false;
+  for (auto aa : rr.byAst(byRefParent).associatedActions()) {
+    if (aa.action() == AssociatedAction::ENTER_CONTEXT) {
+      found = true;
+      assert(aa.fn()->id().symbolPath() == "M.R.enterContext#1");
+    }
+  }
+  assert(found);
+
+  auto byConstRef = findVariable(modules[0], "byConstRef");
+  auto byConstRefParent = parsing::parentAst(ctx, byConstRef)->toAs()->symbol();
+  assert(byConstRefParent);
+  assert(rr.hasAst(byConstRefParent));
+  found = false;
+  for (auto aa : rr.byAst(byConstRefParent).associatedActions()) {
+    if (aa.action() == AssociatedAction::ENTER_CONTEXT) {
+      found = true;
+      assert(aa.fn()->id().symbolPath() == "M.R.enterContext#2");
+    }
+  }
+  assert(found);
+}
+
+static void testNoExplicitImplements() {
+  auto ctx = buildStdContext();
+  ErrorGuard guard(ctx);
+  std::string program =
+    R"""(
+    record R {
+      proc enterContext() {
+        return 42;
+      }
+      proc exitContext(in error: owned Error?) {}
+    }
+    manage new R() as res {}
+    )""";
+
+  auto x = resolveTypesOfVariables(ctx, program, {"res"}).at("res");
+  assert(x.type()->isIntType());
+}
+
+static void testWithoutInterfaceMatch() {
+  auto ctx = buildStdContext();
+  ErrorGuard guard(ctx);
+  std::string program =
+    R"""(
+    record R {
+      proc enterContext() {
+        return 42;
+      }
+      proc exitContext(in err: owned Error?) {} /* wrong name of 'err' */
+    }
+    manage new R() as res {}
+    )""";
+
+  auto x = resolveTypesOfVariables(ctx, program, {"res"}).at("res");
+  assert(x.isUnknownOrErroneous());
+  assert(guard.realizeErrors());
```

Comment:
Might be good to check the class of the error is as expected, and/or that only the `NoMatchingCandidates` error is emitted